Remove WebCertificateInfo

achristensen07 opened the pull request:
#### 30bdacb52588cd7f635adafb027b8253840b5566
<pre>
Remove WebCertificateInfo
<a href="https://bugs.webkit.org/show_bug.cgi?id=245997">https://bugs.webkit.org/show_bug.cgi?id=245997</a>
&lt;rdar://100742393&gt;

Reviewed by Chris Dumez.

* Source/WebKit/Shared/API/c/WKCertificateInfo.cpp:
(WKCertificateInfoGetTypeID):
* Source/WebKit/Shared/API/c/WKCertificateInfo.h:
* Source/WebKit/Shared/API/c/WKSharedAPICast.h:
* Source/WebKit/Shared/UserData.cpp:
(WebKit::UserData::encode):
(WebKit::UserData::decode):
* Source/WebKit/Shared/WebCertificateInfo.h: Removed.
* Source/WebKit/SourcesCocoa.txt:
* Source/WebKit/UIProcess/API/C/WKAuthenticationChallenge.cpp:
* Source/WebKit/UIProcess/API/C/WKContext.cpp:
* Source/WebKit/UIProcess/API/C/WKCredential.cpp:
* Source/WebKit/UIProcess/API/C/mac/WKFrameMac.cpp:
(WKFrameGetServerTrust):
* Source/WebKit/UIProcess/API/C/mac/WKPagePrivateMac.mm:
(-[WKObservablePageState serverTrust]):
* Source/WebKit/UIProcess/API/Cocoa/WKBrowsingContextController.mm:
(-[WKBrowsingContextController certificateChain]):
* Source/WebKit/UIProcess/API/Cocoa/WKNSURLAuthenticationChallenge.mm:
* Source/WebKit/UIProcess/API/Cocoa/WKProcessGroup.mm:
* Source/WebKit/UIProcess/API/Cocoa/WKProcessPool.mm:
* Source/WebKit/UIProcess/API/Cocoa/WKWebView.mm:
(-[WKWebView serverTrust]):
(-[WKWebView _certificateChain]):
(-[WKWebView certificateChain]):
* Source/WebKit/UIProcess/Authentication/AuthenticationChallengeProxy.cpp:
* Source/WebKit/UIProcess/Authentication/AuthenticationDecisionListener.cpp:
* Source/WebKit/UIProcess/Authentication/WebCredential.cpp:
* Source/WebKit/UIProcess/Authentication/WebCredential.h:
(WebKit::WebCredential::create):
* Source/WebKit/UIProcess/Authentication/mac/WebCredentialMac.mm: Removed.
* Source/WebKit/UIProcess/PageLoadState.cpp:
(WebKit::PageLoadState::commitChanges):
(WebKit::PageLoadState::didCommitLoad):
* Source/WebKit/UIProcess/PageLoadState.h:
(WebKit::PageLoadState::certificateInfo const):
* Source/WebKit/UIProcess/WebFrameProxy.cpp:
(WebKit::WebFrameProxy::didCommitLoad):
* Source/WebKit/UIProcess/WebFrameProxy.h:
(WebKit::WebFrameProxy::certificateInfo const):
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::didCommitLoadForFrame):
* Source/WebKit/UIProcess/WebProcessPool.cpp:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebProcess/InjectedBundle/InjectedBundle.h:

Canonical link: <a href="https://commits.webkit.org/255206@main">https://commits.webkit.org/255206@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/4aa52b4937716787bb8dab551cb34d11fcd3855b

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/91591 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/772 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/22203 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/101293 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/161386 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/95596 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/771 "Built successfully") | [  ~~🛠 mac-debug~~](https://ews-build.webkit.org/#/builders/71/builds/29437 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/83934 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/97635 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/97249 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/466 "Passed tests") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/78262 "Built successfully") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/27437 "Found unexpected failure with change (failure)") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/82390 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/82040 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/70475 "Found 292 new API test failures: /WebKitGTK/TestWebKitAccessibility:/webkit/WebKitAccessibility/text/surrogate-pair, /WebKitGTK/TestSSL:/webkit/WebKitWebView/client-side-certificate, /WebKitGTK/TestWebKitWebView:/webkit/WebKitWebView/web-extension-mode, /WebKitGTK/TestWebKitWebView:/webkit/WebKitWebView/is-web-process-responsive, /WebKitGTK/TestWebViewEditor:/webkit/WebKitWebView/select-all/editable, /WebKitGTK/TestWebKitWebView:/webkit/WebKitWebView/save, /WebKitGTK/TestUIClient:/webkit/WebKitWebView/open-window-no-default-size, /WebKitGTK/TestLoaderClient:/webkit/WebKitWebView/load-gresource, /WebKitGTK/TestDOMNode:/webkit/WebKitDOMNode/dom-cache, /TestWebKit:WebKit.GeolocationTransitionToHighAccuracy ... (failure)") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/35714 "Built successfully") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/16070 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/33472 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/17164 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/37309 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/60/builds/39933 "Found 1 new test failure: imported/w3c/web-platform-tests/import-maps/multiple-import-maps/basic.html (failure)") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/1622 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/39215 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/36281 "Passed tests") | | 
<!--EWS-Status-Bubble-End-->

carlosgcampos commented on Source/WebKit/UIProcess/PageLoadState.cpp:
```diff
@@ -103,7 +103,11 @@ void PageLoadState::commitChanges()
+    bool certificateInfoChanged = m_committedState.certificateInfo.trust() != m_uncommittedState.certificateInfo.trust();
+#else
+    bool certificateInfoChanged = false;
+#endif
```
Why is this cocoa specific now?

achristensen07 replied:
yeah, this isn't great.  We were doing pointer comparison of the WebCertificateInfo before, which was not great, and we need a replacement.  We need WebCore::CertificateInfo to have an operator== on all platforms

mcatanzaro replied:
For soup that would be easy, something like this (untested):

```
bool CertificateInfo::operator==(const CertificateInfo& other)
{
    return m_tlsErrors == other.m_tlsErrors && g_tls_certificate_is_same(m_certificate.get(), other->m_certificate.get());
}
```

mcatanzaro replied:
And looks like soup is the only platform that doesn't already have the operator==, so that should do it.

mcatanzaro commented on Source/WebCore/platform/network/soup/CertificateInfo.h:
```diff
@@ -61,6 +61,13 @@ class CertificateInfo {
 
+    bool operator==(const CertificateInfo& other) const
+    {
+        return certificate() == other.certificate()
```
```suggestion
        return g_tls_certificate_is_same(certificate(), other.certificate())
```

mcatanzaro commented on Source/WebCore/platform/network/curl/CertificateInfo.h:
```diff
@@ -57,6 +57,13 @@ class CertificateInfo {
 
     static Certificate makeCertificate(const uint8_t*, size_t);
 
+    bool operator==(const CertificateInfo& other) const
+    {
+        return verificationError() == other.verificationError()
+            && certificateChain() == other.certificateChain();
+    }
+    bool operator!=(const CertificateInfo& other) const { return !(*this == other); }
```
This one already exists as a nonmember function, 10 lines below

cdumez reviewed (approved):
LGTM

Commit:
Remove WebCertificateInfo
https://bugs.webkit.org/show_bug.cgi?id=245997
<rdar://100742393>

Reviewed by Chris Dumez.

* Source/WebKit/Shared/API/c/WKCertificateInfo.cpp:
(WKCertificateInfoGetTypeID):
* Source/WebKit/Shared/API/c/WKCertificateInfo.h:
* Source/WebKit/Shared/API/c/WKSharedAPICast.h:
* Source/WebKit/Shared/UserData.cpp:
(WebKit::UserData::encode):
(WebKit::UserData::decode):
* Source/WebKit/Shared/WebCertificateInfo.h: Removed.
* Source/WebKit/SourcesCocoa.txt:
* Source/WebKit/UIProcess/API/C/WKAuthenticationChallenge.cpp:
* Source/WebKit/UIProcess/API/C/WKContext.cpp:
* Source/WebKit/UIProcess/API/C/WKCredential.cpp:
* Source/WebKit/UIProcess/API/C/mac/WKFrameMac.cpp:
(WKFrameGetServerTrust):
* Source/WebKit/UIProcess/API/C/mac/WKPagePrivateMac.mm:
(-[WKObservablePageState serverTrust]):
* Source/WebKit/UIProcess/API/Cocoa/WKBrowsingContextController.mm:
(-[WKBrowsingContextController certificateChain]):
* Source/WebKit/UIProcess/API/Cocoa/WKNSURLAuthenticationChallenge.mm:
* Source/WebKit/UIProcess/API/Cocoa/WKProcessGroup.mm:
* Source/WebKit/UIProcess/API/Cocoa/WKProcessPool.mm:
* Source/WebKit/UIProcess/API/Cocoa/WKWebView.mm:
(-[WKWebView serverTrust]):
(-[WKWebView _certificateChain]):
(-[WKWebView certificateChain]):
* Source/WebKit/UIProcess/Authentication/AuthenticationChallengeProxy.cpp:
* Source/WebKit/UIProcess/Authentication/AuthenticationDecisionListener.cpp:
* Source/WebKit/UIProcess/Authentication/WebCredential.cpp:
* Source/WebKit/UIProcess/Authentication/WebCredential.h:
(WebKit::WebCredential::create):
* Source/WebKit/UIProcess/Authentication/mac/WebCredentialMac.mm: Removed.
* Source/WebKit/UIProcess/PageLoadState.cpp:
(WebKit::PageLoadState::commitChanges):
(WebKit::PageLoadState::didCommitLoad):
* Source/WebKit/UIProcess/PageLoadState.h:
(WebKit::PageLoadState::certificateInfo const):
* Source/WebKit/UIProcess/WebFrameProxy.cpp:
(WebKit::WebFrameProxy::didCommitLoad):
* Source/WebKit/UIProcess/WebFrameProxy.h:
(WebKit::WebFrameProxy::certificateInfo const):
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::didCommitLoadForFrame):
* Source/WebKit/UIProcess/WebProcessPool.cpp:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebProcess/InjectedBundle/InjectedBundle.h:

Canonical link: https://commits.webkit.org/255206@main
